cloudzap/logger.go

```diff
@@ -24,10 +24,14 @@ func NewLogger(config LoggerConfig) (*zap.Logger, error) {
 		zapConfig := zap.NewDevelopmentConfig()
 		zapConfig.EncoderConfig.EncodeLevel = zapcore.LowercaseColorLevelEncoder
 		zapConfig.Level = zap.NewAtomicLevelAt(config.Level)
-		return zapConfig.Build(
+		logger, err := zapConfig.Build(
 			zap.AddCaller(),
 			zap.AddStacktrace(zap.FatalLevel), // add stacktraces manually where needed
 		)
+		if err == nil {
+			setSlogDefault(logger.Core())
+		}
+		return logger, err
```

Comment:
Understand this is matching the previous change (returning both values), but would probably prefer the more conventional err check 😄 
```suggestion
		if err != nil {
			return nil, fmt.Errorf("init logger: %w", err)
		}
		setSlogDefault(logger.Core())
		return logger, nil
```